src/sentry/static/sentry/app/components/group/sentryAppExternalIssueForm.tsx

```diff
@@ -113,7 +113,13 @@ export class SentryAppExternalIssueForm extends React.Component<Props, State> {
     async (field: FieldFromSchema, input, resolve) => {
       const choices = await this.makeExternalRequest(field, input);
       const options = choices.map(([value, label]) => ({value, label}));
-      return resolve({options});
+      // const options = choices;
+      const optionsByField = new Map(this.state.optionsByField);
+      optionsByField.set(field.name, options);
+      this.setState({
+        optionsByField,
+      });
+      return resolve(options);
```

Comment:
Is `resolve` needed? since this function is `async`, you can just return options I think. Also is it intentional we change the return type? Previously was an object with `options` key, and now it's an array.

Reply:
@billyvg I did not mean to change the return type here, thanks for pointing that out. I guess technically we don't need `getOptions` to return a promise that returns the options because we are passing the options down as a prop. But I think we should just keep it in case we can ever remove the hack.

Reply:
Actually, the change in the return type was necessary because we upgraded the version of React Select.

Reply:
Oops, misread this re: `resolve`, did not see it was passed through getOptions